pkg/controller/deployment/deployment_controller.go

```diff
@@ -774,10 +774,10 @@ func copyDeploymentAnnotationsToReplicaSet(deployment *extensions.Deployment, rs
 		rs.Annotations = make(map[string]string)
 	}
 	for k, v := range deployment.Annotations {
-		// Skip apply annotations
+		// Skip apply annotations and revision annotations
 		// TODO: How to decide which annotations should / should not be copied?
 		// See https://github.com/kubernetes/kubernetes/pull/20035#issuecomment-179558615
-		if k == kubectl.LastAppliedConfigAnnotation || rs.Annotations[k] == v {
+		if k == kubectl.LastAppliedConfigAnnotation || k == deploymentutil.RevisionAnnotation || rs.Annotations[k] == v {
```

Comment:
The deployment's revision number is updated from checking its new RS revision, but not the other way around, since new rs revision is updated automatically from calculating the deployment's RSes revisions. 

We shouldn't update rs's revision from its deployment, since it's possible that we update the new RS's revision, and then failed trying to update the deployment's revision.